### ISSUE TITLE
Completions that are effectively code-generated overrides aren't inserted properly

### ISSUE THREAD

**Issue by SCWells72** [redhat-developer/lsp4ij#685]
Some LSP-provided completions are intended to behave as code generation, e.g., implementing/overriding a method from an interface/base class. When I accepted such a completion, it resulted in invalid code, e.g., `HTMLElement.click()` in JavaScript/TypeScript which results in:

```typescript
click(): void {
    $0
\}()
```

Here's a demonstration of the issue:

![Image](https://github.com/user-attachments/assets/e1800125-bb35-4f02-9188-e4a4781ca088)

It looks like it's just a current limited handling of templatized completion results.

**Comment by angelozerr** [comment]
Are you sure this completion comes from language server and not from IJ himself?

Could you check in lsp completion returns this click snippet.

**Comment by angelozerr** [comment]
If it comes from language server, could yiu copy paste the lsp response in the issue please.

**Comment by SCWells72** [comment]
Yes, it's coming from the TypeScript language server. Here's the response to `textDocument/completion` (retaining only the relevant completion):

```json
{
  "isIncomplete": false,
  "items": [
    {
      "label": "click",
      "kind": 2,
      "sortText": "11",
      "filterText": "click(): void {\n    $0\n}",
      "insertText": "click(): void {\n    $0\n}",
      "insertTextFormat": 2,
      "data": {
        "cacheId": 87
      }
    }
  ]
}
```

and here's the request/response for `completionItem/resolve` for that completion:

**Request**
```json
{
  "label": "click",
  "kind": 2,
  "sortText": "11",
  "filterText": "click(): void {\n    $0\n}",
  "insertText": "click(): void {\n    $0\n}",
  "insertTextFormat": 2,
  "data": {
    "cacheId": 87
  }
}
```

**Response**
```json
{
  "label": "click",
  "kind": 2,
  "detail": "(method) HTMLElement.click(): void",
  "documentation": {
    "kind": "markdown",
    "value": "Called when the element is inserted in a document"
  },
  "sortText": "11",
  "filterText": "click(): void {\n    $0\n}",
  "insertText": "click(): void {\n    \\$0\n\\}()$0",
  "insertTextFormat": 2,
  "data": {
    "file": "/path/to/sourceFile.ts",
    "line": 41,
    "offset": 9,
    "entryNames": [
      "click"
    ]
  }
}
```

To be honest, the `insertText` from the detailed response looks suspicious to me since it would have parentheses after the method body close brace. I'm not sure what `"insertTextFormat": 2` implies either, but presumably some type of guidance on how to process the "template" from the insert text.

**Comment by angelozerr** [comment]
@SCWells72 I suggest that you create an issue in typescript-language-server repo.

**Comment by SCWells72** [comment]
> [@SCWells72](https://github.com/SCWells72) I suggest that you create an issue in typescript-language-server repo.

Pretty sure it's an issue with LSP4IJ specifically. Here's the behavior in VS Code:

![Image](https://github.com/user-attachments/assets/3a382b1e-4d27-496f-bada-47ffa3712d23)

and here's the behavior in LSP4IJ:

![Image](https://github.com/user-attachments/assets/818c1a9c-028c-43bc-bb8b-1be4386bfa19)

**Comment by angelozerr** [comment]
vscode don't use typescript-language-server but directly tssserver.

When I see your trace it seems that the resolve response add \\ which escape the snippet.
